airbyte-integrations/connectors/source-notion/unit_tests/test_streams.py

```diff
@@ -76,3 +77,61 @@ def test_backoff_time(patch_base_class):
     stream = NotionStream(config=MagicMock())
     expected_backoff_time = None
     assert stream.backoff_time(response_mock) == expected_backoff_time
+
+
+def test_users_request_params(patch_base_class):
+    stream = Users(config=MagicMock())
+
+    # No next_page_token. First pull
+    inputs = {"stream_slice": None, "stream_state": None, "next_page_token": None}
+    expected_params = {"page_size": 100}
+    assert stream.request_params(**inputs) == expected_params
+
+    # When getting pages after the first pull.
+    inputs = {"stream_slice": None, "stream_state": None, "next_page_token": {"next_cursor": "123"}}
+    expected_params = {"start_cursor": "123", "page_size": 100}
+    assert stream.request_params(**inputs) == expected_params
+
+
+def test_user_stream_handles_pagination_correclty(requests_mock):
+    """
+    Test shows that Users stream uses pagination as per Notion API docs.
+    """
+
+    response_body = {
+        "object": "list",
+        "results": [{"id": f"{x}"} for x in range(100)],
+        "next_cursor": "test_cursor_1",
+        "has_more": True,
+        "type": "user"
+    }
+    url = "https://api.notion.com/v1/users?page_size=100"
```

Comment:
Is there a need to declare the **_url_** variable? couldn't the url be passed directly to **_requests_mock.get()_**?

Reply:
Moved to `requests_mock.get()` in [fda250e](https://github.com/airbytehq/airbyte/pull/11452/commits/fda250ecf078624eff22eb745c3eaa7337c947ce).